platform/feature-api/api/rdf-spec/src/data-factory.ts

```diff
@@ -627,22 +673,27 @@ export class DataFactorySpec<OutQuad extends BaseQuad = Quad> {
                         it(key, () => {
                             fc.assert(
                                 fc.property(g, (term) => {
+                                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
                                     assert((term as any).equals(term));
                                 })
                             );
                         });
             });
+
             describe("not equal (terms)", () => {
                 const keys = ["namedNode", "blankNode", "literal", "variable"];
                 for (const key1 of keys)
                     for (const key2 of keys)
                         if (key1 !== key2)
                             it(`${key1}/${key2}`, () => {
+                                // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
OK, this is tricky, but I would say that gen will have one of those 4 strings as attributes, is that correct? Or that they are simply an `array[string]`, right?

Reply:
Hmm not sure. `keyN` here are strings inded (of this array above -line#684) but `gen` here has the type `Gens<OutQuad>`. so this casting that is tried here, indeed is needed in order to retrieve the `key2` index.
(But I don't understand the code's intention here really well tbh - so preferred to not touch it atm)